src/cdi_ndi/NDI_CP_Eloss.cc

```diff
@@ -0,0 +1,288 @@
+//----------------------------------*-C++-*-----------------------------------//
+/*!
+ * \file   cdi_ndi/NDI_CP_Eloss.cc
+ * \author Ben R. Ryan
+ * \date   2020 Jun 3
+ * \brief  NDI_CP_Eloss member definitions.
+ * \note   Copyright (C) 2019-2020 Triad National Security, LLC.
+ *         All rights reserved. */
+//----------------------------------------------------------------------------//
+
+#include "NDI_CP_Eloss.hh"
+#include "ds++/DracoStrings.hh"
+#include "ds++/Soft_Equivalence.hh"
+
+#include <cmath>
+
+namespace rtt_cdi_ndi {
+
+//------------------------------------------------------------------------------------------------//
+/*!
+ * \brief Do a 3D linear interpolation between vertices of a rectangular prism.
+ *
+ * Algorithm from wikipedia's Trilinear Interpolation article, hat tip to E.
+ * Norris for the reference.
+ *
+ * \param[in] x0   lower x coordinate of lattice
+ * \param[in] x1   upper x coordinate of lattic
+ * \param[in] y0   lower y coordinate of lattice
+ * \param[in] y1   upper y coordinate of lattice
+ * \param[in] z0   lower z coordinate of lattice
+ * \param[in] z1   upper z coordinate of lattic
+ * \param[in] f000 function at (x0,y0,z0)
+ * \param[in] f100 function at (x1,y0,z0)
+ * \param[in] f001 function at (x0,y0,z1)
+ * \param[in] f101 function at (x1,y0,z1)
+ * \param[in] f010 function at (x0,y1,z0)
+ * \param[in] f110 function at (x1,y1,z0)
+ * \param[in] f011 function at (x0,y1,z1)
+ * \param[in] f111 function at (x1,y1,z1)
+ * \param[in] x    x coordinate of interpolation point
+ * \param[in] y    y coordinate of interpolation point
+ * \param[in] z    z coordinate of interpolation point
+ * \return The function value linearly interpolated to (x,y,z)
+ */
+inline double linear_interpolate_3(double const x0, double const x1, double const y0,
```

Comment:
This is duplicated from `cdi_CPEloss::Tabular_CP_Eloss`, I'm not sure what the right approach to avoid code duplication is right now.

Reply:
Oh, gotcha! Let me poke around to see if there's a different draco dir we could move this to (for sharing purposes). I think @attom might also have some interpolation functions related to CSK... perhaps they could find a shared home? Just spitballing here, I'll look into it and report back :)

Reply:
@brryan I will plan to move the interpolation code to a shared draco lib after this is merged in - I think Andrew Till has some interpolators in his compton PR that could live in there, too. I'll make myself a ticket :)

Reply:
Sounds good, thanks! If this function ends up being too specific for a really general Draco library (it should really be called `linear_interpolate_uniform_rectangular_prism_lattice` or `LIURPL` or some such) we could just move it to `cdi/CPEloss.hh`

Reply:
Haha, I like the sound of LIURPL!

Reply:
That may already be a pokemon, now that I think about it